cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperScaler.java

```diff
@@ -0,0 +1,351 @@
+                            log.debug("Successfully connected to {} to scale Zookeeper", zookeeperConnectionString);
+                            connected.tryComplete();
+                        }
+                        break;
+                    default:
+                        if (!connected.future().isComplete()) {
+                            log.warn("Failed to connect to {} to scale Zookeeper because of state {}", zookeeperConnectionString, watchedEvent.getState());
+                            closeConnectionSync();
+                            connected.tryFail(new RuntimeException("Failed to connect to Zookeeper " + zookeeperConnectionString + " - got state " + watchedEvent.getState()));
+                        }
+                }
+
+                vertx.cancelTimer(timer);
+            }, getClientConfig());
+        } catch (IOException e)   {
+            log.warn("Failed to connect to {} to scale Zookeeper", zookeeperConnectionString, e);
+            connected.tryFail(new RuntimeException("Failed to connect to Zookeeper " + zookeeperConnectionString, e));
+        }
+
+        return connected.future();
+    }
+
+    /**
+     * Internal method to scale Zookeeper up or down or check configuration. It will:
+     *     1) Compare the current configuration with the desired configuration
+     *     2) Update the configuration if needed
+     *
+     * @param currentServers    Current configuration
+     * @param scaleTo           Desired scale
+     * @return                  Future indicating success or failure
+     */
+    private Future<Void> scaleTo(Map<String, String> currentServers, int scaleTo) {
+        Map<String, String> desiredServers = generateConfig(scaleTo);
+
+        if (isDifferent(currentServers, desiredServers))    {
+            log.debug("The Zookeeper server configuration needs to be updated");
+            return updateConfig(desiredServers).map((Void) null);
+        } else {
+            log.debug("The Zookeeper server configuration is already up to date");
+            return Future.succeededFuture();
+        }
+    }
+
+    /**
+     * Gets the current configuration from Zookeeper.
+     *
+     * @return  Future containing Map with the current Zookeeper configuration
+     */
+    private Future<Map<String, String>> getConfig()    {
+        Promise<Map<String, String>> configPromise = Promise.promise();
+
+        vertx.executeBlocking(promise -> {
+            try {
+                byte[] config = zkAdmin.getConfig(false, null);
+                Map<String, String> servers = parseConfig(config);
+                log.debug("Current Zookeeper configuration is {}", servers);
+                promise.complete(servers);
+            } catch (KeeperException | InterruptedException e)    {
+                log.warn("Failed to get current Zookeeper server configuration", e);
+                promise.fail(new RuntimeException("Failed to get current Zookeeper server configuration", e));
```

Comment:
Probably worth having a `ZookeeperScalingException` if these can end up in the `status` of the CR.